Pipfile

```diff
@@ -6,6 +6,9 @@ name = "pypi"
 [packages]
 click = "*"
 docker = "*"
+duckdb = "*"
+ijson = "*"
+pyarrow = "*"
```

Comment:
Commenting here, but would be a `dev package.

Might be worth installing `pyarrow-stubs`?  In some local testing, I've found it helpful to expose to raise linting errors when types might be functional, but not ideal.

Example from `collate_ab_transforms.py`:
```python
TRANSFORMED_DATASET_SCHEMA = pa.schema(
    [
        pa.field("timdex_record_id", pa.string()),
        pa.field("source", pa.string()),
        pa.field("record", pa.binary()),
        pa.field("version", pa.string()),
        pa.field("transformed_file_name", pa.string()),
    ]
)
JOINED_DATASET_SCHEMA = pa.schema(
    [
        pa.field("timdex_record_id", pa.string()),
        pa.field("source", pa.string()),
        pa.field("record_a", pa.binary()),
        pa.field("record_b", pa.binary()),
    ]
)
```

As is, this will raise a linting error where `pa.schema` cannot accept a list.  But converting them to a tuple works!